vignettes/ChromBand.Rmd

```diff
@@ -0,0 +1,437 @@
+##                         which(cyt2$chrom == "12"))
+## cyt2[c12p12_idx, "gieStain"] <- rep(c("gpos50", "gpos75"),
+##                                     length=length(c12p12_idx))
+##
+##
+## plotCytoband2(chromosome="12", build=build, cytoband=cyt2, outer=FALSE,  ## From SNPchip
+##     cex.axis=0.6, main="Human chromosome 12")
+## plotCytoband2(chromosome="12", build=build, cytoband=cyt2, outer=FALSE,
+##               cex.axis=0.6,
+##               main="Human chromosome 12")
+
+```
+<figcaption>Ideogram for human chromosome 12. The p arm is on the left, the q arm is on the right, and the centromere is indicated by a notch. The shaded bands together represent 12q21. This band is composed of three sub-bands: 12q21.1, 12q21.2, and 12q21.3. The last of these is composed of sub-sub-bands 12q21.31, 12q21.32, and 12q21.33.</figcaption>
+</figure>
+
+# Data Preparation
+
+For illustration, we use a microarray dataset [@Chiaretti2005] from a clinical trial in acute lymphoblastic leukemia (ALL). The data are described in Chapter 2 of [@BiocCaseStudies]. For the analysis presented here, we consider the comparison of two subsets: patients identified as having a BCR/ABL gene fusion present, typically as a result of a translocation of chromosomes 9 and 22 (labeled BCR/ABL), and those that have no observed cytogenetic abnormalities (labeled NEG). The full dataset is available in the [ALL]{.sans-serif} package, and the relevant subset of the data can be obtained by
+
+```{r bcrAblOrNegSubset, cache=FALSE}
+
+data(ALL, package="ALL")
+
+subsetType <- "BCR/ABL"
+
+Bcell <- grep("^B", as.character(ALL$BT) )
+
+bcrAblOrNegIdx <- which(as.character(ALL$mol.biol) %in% c("NEG", subsetType) )
+
+bcrAblOrNeg <- ALL [, intersect(Bcell, bcrAblOrNegIdx) ]
+
+bcrAblOrNeg$mol.biol <- factor(bcrAblOrNeg$mol.biol)
+
+```
+
+We also create relevant annotation maps to go from feature names to Entrez ID, gene symbol, and chromosome band.
+
+```{r annMaps, cache=FALSE}
+
+annType <- c("db", "env")
+
+entrezMap <- getAnnMap("ENTREZID", annotation(bcrAblOrNeg),
+                       type=annType, load=TRUE)
+
+symbolMap <- getAnnMap("SYMBOL", annotation(bcrAblOrNeg),
+                       type=annType, load=TRUE)
+
+bandMap <- getAnnMap("MAP", annotation(bcrAblOrNeg),
+                     type=annType, load=TRUE)
+
+```
+
+We applied a non-specific filter to the dataset to remove probesets lacking the desired annotation as well as those with an interquartile range (IQR) below the median IQR, as probesets with little variation across samples are uninformative. We also ensured that each Entrez Gene identifier maps to exactly one probeset by selecting the probeset with the largest IQR when two or more probesets map to the same Entrez Gene ID.
+
+```{r nsFiltering}
+
+filterAns <- nsFilter(bcrAblOrNeg,
+                     require.entrez = TRUE,
+                     remove.dupEntrez = TRUE,
+                     var.func = IQR, var.cutoff = 0.5)
```

Comment:
Line up the arguments:

    filterAns <- nsFilter(bcrAblOrNeg,
                          require.entrez = TRUE,
                          remove.dupEntrez = TRUE,
                          var.func = IQR, var.cutoff = 0.5)